### PR TITLE
Replaced use of Junit TemporaryFolder with Files.createTempDirectory

### PR DESCRIPTION
**Pull request by jzgithub1**
in a few tests to prevent random test failures.  This was recommended on the Junit4 thread to fix this temporary issue.  This code change has been tested on a single machine and the code builds with:
'mvn -Pdev -Ddeploy -Dtar  clean install'

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/tables/IndexQueryLogicTest.java [hunk 1]
```diff
@@ -60,7 +61,7 @@ public static void setupClass() throws Exception {
     }
     
     @Before
-    public void querySetUp() {
+    public void querySetUp() throws IOException {
```
Comment (ivakegg) [inline warehouse/query-core/src/test/java/datawave/query/tables/IndexQueryLogicTest.java]: Why adding this throws?

Comment (ivakegg) [reply warehouse/query-core/src/test/java/datawave/query/tables/IndexQueryLogicTest.java]: Oh, nevermind.  I see.

**Comment by jzgithub1** [comment]
Because Files.createTempDirectory throws and IOException

File: warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java [hunk 2]
```diff
@@ -139,7 +138,9 @@ protected ShardQueryLogic createQueryLogic() {
     }
     
     @Before
-    public void querySetUp() {
+    public void querySetUp() throws IOException {
+        temporaryFolder = Files.createTempDirectory("tmp");
```
Comment (brianloss) [inline warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java]: Does this folder ever get cleaned up?

Comment (jzgithub1) [reply warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java]: Thanks for spotting that.  I need an @After function where I do that.  I did this in two other classes so it got a little confusing.

Comment (jzgithub1) [reply warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java]: It seemed get wiped out anyways.

Comment (jzgithub1) [reply warehouse/query-core/src/test/java/datawave/query/testframework/AbstractFunctionalQuery.java]: Pushed the code fix. I hit the ' temporaryFolder.toFile().deleteOnExit();' line when I put a breakpoint on it in the Intellij debugger and run IndexQueryLogicTest which uses AbstractFunctionalQuery.

**Pull request merged**